src/net/dns.rs

```diff
@@ -631,8 +620,13 @@ pub(crate) async fn lookup_host_with_cache(
             }
         }
 
-        if resolved_addrs.is_empty() {
-            return Ok(load_hardcoded_cache(hostname, port));
+        if let Some(ips) = DNS_PRELOAD.get(hostname) {
```

Comment:
It seems that this cache is safe to use even if load_cache is false, i.e. for not authenticated connections. Anyway this doesn't block merging the PR